### PR TITLE
OCM-16: ability to set worker disk size

### PR DESCRIPTION
**Pull request by leseb**
This is a draft PR linked to the pending backend change here: https://gitlab.cee.redhat.com/service/uhc-clusters-service/-/merge_requests/5790.

I'm opening this early PR for feedback as I'm not familiar with this code base.
Thanks for the early review.

### CODE REVIEW

**Comment by leseb** [comment]
@bardielle PTAL

File: provider/cluster_rosa_classic_resource.go [hunk 1]
```diff
@@ -1844,6 +1891,35 @@ func proxyValidators() []tfsdk.AttributeValidator {
 	}
 }
 
+func rootVolumeValidators() []tfsdk.AttributeValidator {
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:1894]: :+1:

File: provider/cluster_rosa_classic_resource.go [hunk 2]
```diff
@@ -1844,6 +1891,35 @@ func proxyValidators() []tfsdk.AttributeValidator {
...
+				if rootVolumeSize < 0 {
+					resp.Diagnostics.AddError(errSum, "Expected root volume size to be greater than 0")
+					return
+				}
+			},
+		},
+	}
+}
+
 func propertiesValidators() []tfsdk.AttributeValidator {
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:1923]: You should also add the assignment of those properties in the "Create" function
Also, if this attribute (compute_machine_type_state) can be updated and modified, please add the support to the "update" function.

**Comment by bardielle** [comment]
@leseb please add subsytem tests for cluster rosa resource and machine

File: provider/cluster_rosa_classic_state.go [hunk 3]
```diff
@@ -21,45 +21,47 @@ import (
...
+	AWSPrivateLink     types.Bool   `tfsdk:"aws_private_link"`
+	Sts                *Sts         `tfsdk:"sts"`
+	CCSEnabled         types.Bool   `tfsdk:"ccs_enabled"`
+	EtcdEncryption     types.Bool   `tfsdk:"etcd_encryption"`
+	AutoScalingEnabled types.Bool   `tfsdk:"autoscaling_enabled"`
+	MinReplicas        types.Int64  `tfsdk:"min_replicas"`
+	MaxReplicas        types.Int64  `tfsdk:"max_replicas"`
+	ChannelGroup       types.String `tfsdk:"channel_group"`
+	CloudRegion        types.String `tfsdk:"cloud_region"`
+	// ComputeMachineType should have been named ComputeMachineTypeID
```
[the remark this answers is not in the document]
Comment (leseb) [reply provider/cluster_rosa_classic_state.go:36]: If it's not breaking anything then yes :)

**Comment by leseb** [comment]
> @leseb please add subsytem tests for cluster rosa resource and machine

Yes yes it's planned, I just wanted to know first if I was heading in the right direction, thanks for the prompt review!

File: provider/cluster_rosa_classic_state.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bardielle) [reply provider/cluster_rosa_classic_state.go:36]: It should be aligned with Rosa cli

**Comment by leseb** [comment]
Closing in favour of https://github.com/terraform-redhat/terraform-provider-rhcs/pull/244

**Pull request closed without merging**